src/clj/web/game.clj

```diff
@@ -1,6 +1,6 @@
 (ns web.game
   (:require [web.ws :as ws]
-            [web.lobby :refer [all-games old-states] :as lobby ]
+            [web.lobby :refer [all-games old-states] :as lobby]
```

Comment:
Hey that space was important to me!

Reply:
Thanks Neal.   The other other way would be to store incremental diff but I think that is prone to a lot of issues. Or the data could be stored in app-state on the clients.

Reply:
Would add - I can (dissoc) the :log key from the state before saving it - as you rightly say we don't want to over-write the chat log to we don't need to keep it.  We can grab it from the existing state atom.  This is probably a big chunk saved.  

Also the other big(-ish) key would usually be :turn-events which will be blank for this. 

Finally noticed a bit of cleaning up to be done from the stats implementation.  Game stats are kept in every message in the log right now.  Who the hell did that work?
       {:user {:_id "5a91b8d9f736882837e03f41",
               :username "wozzit",
               :emailhash "201c6d26d16065a4f678b8340c4625ec",
               :options {},
               :stats {:games-started 69,
                       :games-started-runner 68,
                       :games-completed 48,
                       :games-completed-runner 47,
                       :games-started-corp 1,
                       :games-completed-corp 1}},
        :text "sfd"}

Reply:
This looks useful to see how much space the atom is using.  http://clojure-goes-fast.com/blog/introspection-tool-object-memory-meter/   Will play around on dev... my feel is the game-state is tiny compared to all the other stuff running on the JVM

Reply:
Brand new game with no :turn-state added or any user chat:
```(clj-memory-meter.core/measure state)
=> "183.1 KB"```

With turn started, whole game-state grafted into the :turn-state key:
```(clj-memory-meter.core/measure state)
=> "187.8 KB"```

At end of corp turn 1 having taken 3 actions (install, play action, play action).  No user chat.  Noting this will add a bunch of data to the :turn-events and :log keys.
```(clj-memory-meter.core/measure state)
=> "195.1 KB"```

Runner hits Start Turn: - noting this includes :log from the previous turn still in this build.
```(clj-memory-meter.core/measure state)
=> "197.0 KB"```

One of the players says "I AM A PONY!!" x 10...
```(clj-memory-meter.core/measure state)
=> "199.4 KB"```

Runner does 4 actions then ends-turn:
```(clj-memory-meter.core/measure state)
=> "206.3 KB"```

Corp hits "Start turn"
```(clj-memory-meter.core/measure state)
=> "210.8 KB"```

And probably the most useful.  Reset! the atom at this point by removing the :turn-state data:
```(reset! state (dissoc @state :turn-state))

(clj-memory-meter.core/measure state)
=> "196.1 KB"```

So the diff is ~10-15% vs. current usage.  Likely a bit lower once we drop the 👍 log key

I am thinking to combine the turn undo and click undo - so we only allow one rollback.  Thoughts?

Reply:
Right, the miracles of persistent hashmap data structures. Looking great. Full steam ahead!

Reply:
Should be good to merge.  I could not yet find a reason to allow replaying back multiple turns but not hard to add if we want to. 

Undo-click will be via another PR.  Actually harder... as you can spend clicks in multiple ways in the engine.